### ISSUE TITLE
False positive `A003` when overriding `threading.Event.set`

### ISSUE THREAD

**Issue by jamesbraza** [astral-sh/ruff#5956]
```python
from threading import Event

class MyEvent(Event):
    def set(self) -> None:
        super().set()
        print("My override")
```

Running `ruff==0.0.279`:

```bash
> ruff --isolated --select="A003" a.py
a.py:5:9: A003 Class attribute `set` is shadowing a Python builtin
```

This seems like a false positive, as one has to use the word `set` to override the `Event` parent class's method.

Should `A003` be raised on method overrides?

**Comment by charliermarsh** [comment]
Probably not... but somewhat low-priority for me, since we can't resolve the superclasses anyway and it's not a hugely popular rule. The most lightweight thing I could see us doing for now is ignoring methods annotated with `@overrides`.

**Comment by jamesbraza** [comment]
Just out of curiosity, why can't `ruff` resolve superclasses?  I believe other linters like `pylint` have this capability

**Comment by charliermarsh** [comment]
Ruff doesn't resolve symbols across files right now, though it will in the future (I've been working on it a bit but there's no defined timeline for it).

`pylint` _does_ resolve symbols across files and does more static analysis than other linters, though not all linters function that way -- Flake8 (as an alternative example) couldn't support this.
